packages/by-name/kata/kata-runtime/0013-genpolicy-allow-image_guest_pull.patch

```diff
@@ -0,0 +1,1471 @@
+     overlay_storage.driver == "overlayfs"
+-    print("allow_storages: overlay_storage =", overlay_storage)
++    print("allow_storages 2: overlay_storage =", overlay_storage)
+     count(overlay_storage.options) == 2
+ 
+     layer_ids := split(overlay_storage.options[0], ":")
+-    print("allow_storages: layer_ids =", layer_ids)
++    print("allow_storages 2: layer_ids =", layer_ids)
+ 
+     root_hashes := split(overlay_storage.options[1], ":")
+-    print("allow_storages: root_hashes =", root_hashes)
++    print("allow_storages 2: root_hashes =", root_hashes)
+ 
+     every i_storage in i_storages {
+         allow_storage(p_storages, i_storage, bundle_id, sandbox_id, layer_ids, root_hashes)
+     }
+ 
+-    print("allow_storages: true")
++    print("allow_storages 2: true")
+ }
+ 
+ allow_storage(p_storages, i_storage, bundle_id, sandbox_id, layer_ids, root_hashes) {
+diff --git a/src/tools/genpolicy/src/policy.rs b/src/tools/genpolicy/src/policy.rs
+index fb1e92388..2416832c3 100644
+--- a/src/tools/genpolicy/src/policy.rs
++++ b/src/tools/genpolicy/src/policy.rs
+@@ -270,6 +270,9 @@ pub struct ContainerPolicy {
+     /// Data compared with req.OCI for CreateContainerRequest calls.
+     pub OCI: KataSpec,
+ 
++    /// Container image reference.
++    image: String,
++
+     /// Data compared with req.storages for CreateContainerRequest calls.
+     storages: Vec<agent::Storage>,
+ 
+@@ -638,6 +641,7 @@ impl AgentPolicy {
+                 Annotations: annotations,
+                 Linux: linux,
+             },
++            image: yaml_container.image.clone(),
+             storages,
+             devices,
+             sandbox_pidns,
+diff --git a/src/tools/genpolicy/tests/main.rs b/src/tools/genpolicy/tests/main.rs
+index 3b41092f0..3abe26d23 100644
+--- a/src/tools/genpolicy/tests/main.rs
++++ b/src/tools/genpolicy/tests/main.rs
+@@ -104,3 +104,8 @@ fn test_create_sandbox() {
+ fn test_create_container_process() {
+     runtests::<CreateContainerRequest>("createcontainer/deploy_process");
+ }
++
++#[test]
++fn test_create_container_guest_pull() {
++    runtests::<CreateContainerRequest>("createcontainer/guest_pull");
++}
+diff --git a/src/tools/genpolicy/tests/testdata/createcontainer/guest_pull/pod.yaml b/src/tools/genpolicy/tests/testdata/createcontainer/guest_pull/pod.yaml
+new file mode 100644
+index 000000000..203af19a6
+--- /dev/null
++++ b/src/tools/genpolicy/tests/testdata/createcontainer/guest_pull/pod.yaml
+@@ -0,0 +1,11 @@
++apiVersion: v1
++kind: Pod
++metadata:
++  name: dummy
++spec:
++  runtimeClassName: kata-cc-isolation
++  containers:
++  - name: dummy-ctr
++    image: registry.k8s.io/pause:3.6@sha256:3d380ca8864549e74af4b29c10f9cb0956236dfb01c40ca076fb6c37253234db
++    securityContext:
++      privileged: true
+diff --git a/src/tools/genpolicy/tests/testdata/createcontainer/guest_pull/testcases.json b/src/tools/genpolicy/tests/testdata/createcontainer/guest_pull/testcases.json
```

Comment:
I think I'll factor out the test changes.

Reply:
Decided against it so that the patches here stay closer to what is going to be backported.